storybookjs/storybook
# Angular:  Fix `sb init` library support

Closes #21657
Closes #21672

## What I did
* [x] add template files for angular library projects, specifically `tsconfig.json` to reference `tsconfig.lib.json`.
* [x] select template files based on `projectType` from `angular.json`.
* [x] update `stories` path in `main.ts` to include `../src` directory if found as a relative path of `.storybook` instead of relative path of `.`
* [x] return `configDir` as a property of `installResult` to use with `automigrate`.

## How to test

* [x] create a new angular workspace `npx ng new workspace --no-create-application`
* [x] create a new library inside the new workspace `cd workspace` then `npx ng g library my-lib`
* [x] execute `npx sb@next init`

Verify the following: 
* [x] `automigrate` runs automatically at the end of `sb init` with the following output
```shell
╭ No migrations were applicable to your project ───────────────────────────────────────────╮
│                                                                                          │
│   If you'd like to run the migrations again, you can do so by running 'npx               │
│   storybook@next automigrate'                                                            │
│                                                                                          │
│   The automigrations try to migrate common patterns in your project, but might not       │
│   contain everything needed to migrate to the latest version of Storybook.               │
│                                                                                          │
│   Please check the changelog and migration guide for manual migrations and more          │
│   information: https://storybook.js.org/migration-guides/7.0                             │
│   And reach out on Discord if you need help: https://discord.gg/storybook                │
│                                                                                          │
╰──────────────────────────────────────────────────────────────────────────────────────────╯
```
instead of seeing the following error 
```shell
[Storybook automigrate] Could not find or evaluate your Storybook main.js config directory at .storybook so the automigrations will be skipped. You might be running this command in a monorepo or a non-standard project structure. If that is the case, please rerun this command by specifying the path to your Storybook config directory via the --config-dir option.
```
* [x] `workspace/projects/my-lib/.storybook/tsconfig.json` extends `tsconfig.lib.json` instead of `tsconfig.app.json`
* [x] `workspace/projects/my-lib/.storybook/main.ts` has stories with prefix `../src/` instead of `../stories/`
